### DISCUSSION TITLE
blog/analyze-memory-leaks-in-your-nuxt-app

### DISCUSSION THREAD

**Discussion by giscus** [Mokkapps/mokkapps-de-discussions#16] [Announcements]
# blog/analyze-memory-leaks-in-your-nuxt-app

In one of my client projects, we recently had to analyze and fix a memory leak in our Nuxt 3+ application. I share my experience and our steps to identify and fix the memory leak.

https://mokkapps.de/blog/analyze-memory-leaks-in-your-nuxt-app

**Comment by chibx** [comment]
Thanks a lot Michael, this was helpful a lot.
I wasn't even aware that a **watch** after an await could cause such leaks 😅. I would like to know if it's generally safe to use watchers before awaits, or in fact outside of the onMounted hook when working with SSR

**Reply by Mokkapps** [reply]
I think so, there also exist ESLint rules: https://eslint.vuejs.org/rules/no-watch-after-await.html
